go/libraries/doltcore/schema/typeinfo/uint.go

```diff
@@ -0,0 +1,294 @@
+			return types.Uint(val), nil
+		case int64:
+			return types.Uint(val), nil
+		case uint:
+			return types.Uint(val), nil
+		case uint8:
+			return types.Uint(val), nil
+		case uint16:
+			return types.Uint(val), nil
+		case uint32:
+			return types.Uint(val), nil
+		case uint64:
+			return types.Uint(val), nil
+		case float32:
+			return types.Uint(val), nil
+		case float64:
+			return types.Uint(val), nil
+		case string:
+			return types.Uint(artifact), nil
+		case types.Null:
+			return types.NullValue, nil
+		case types.Bool:
+			if val {
+				return types.Uint(1), nil
+			}
+			return types.Uint(0), nil
+		case types.Int:
+			return types.Uint(val), nil
+		case types.Uint:
+			return val, nil
+		case types.Float:
+			return types.Uint(val), nil
+		case types.String:
+			return types.Uint(artifact), nil
+		default:
+			return nil, fmt.Errorf(`"%v" has falsely evaluated value "%v" of type "%T" as valid`, ti.String(), val, val)
+		}
+	}
+	return nil, fmt.Errorf(`"%v" cannot convert value "%v" of type "%T" as it is invalid`, ti.String(), v, v)
+}
+
+// Equals implements TypeInfo interface.
+func (ti *uintImpl) Equals(other TypeInfo) bool {
+	if other == nil {
+		return false
+	}
+	if ti2, ok := other.(*uintImpl); ok {
+		return ti.Width == ti2.Width
+	}
+	return false
+}
+
+// GetTypeIdentifier implements TypeInfo interface.
+func (ti *uintImpl) GetTypeIdentifier() Identifier {
+	return UintTypeIdentifier
+}
+
+// GetTypeParams implements TypeInfo interface.
+func (ti *uintImpl) GetTypeParams() map[string]string {
+	return map[string]string{uintTypeParam_Width: strconv.Itoa(int(ti.Width))}
+}
+
+// IsValid implements TypeInfo interface.
+func (ti *uintImpl) IsValid(v interface{}) bool {
+	_, ok := ti.isValid(v)
+	return ok
+}
+
+// NomsKind implements TypeInfo interface.
+func (ti *uintImpl) NomsKind() types.NomsKind {
+	return types.UintKind
+}
+
+// String implements TypeInfo interface.
+func (ti *uintImpl) String() string {
+	switch ti.Width {
+	case UintWidth8:
+		return "Uint8"
+	case UintWidth16:
+		return "Uint16"
+	case UintWidth24:
+		return "Uint24"
+	case UintWidth32:
+		return "Uint32"
+	case UintWidth64:
+		return "Uint64"
+	default:
+		panic(fmt.Errorf(`uint width "%v" is not valid`, ti.Width))
+	}
+}
+
+// ToSqlType implements TypeInfo interface.
+func (ti *uintImpl) ToSqlType() sql.Type {
+	switch ti.Width {
+	case UintWidth8:
+		return sql.Uint8
+	case UintWidth16:
+		return sql.Uint16
+	case UintWidth24:
+		return sql.Uint24
+	case UintWidth32:
+		return sql.Uint32
+	case UintWidth64:
+		return sql.Uint64
+	default:
+		panic(fmt.Errorf(`uint width "%v" is not valid`, ti.Width))
+	}
+}
+
+// isValid is an internal implementation for the TypeInfo interface function IsValid.
+// Some validity checks process the value into its final form, which may be returned
+// as an artifact so that a value doesn't need to be processed twice in some scenarios.
+func (ti *uintImpl) isValid(v interface{}) (artifact uint64, ok bool) {
```

Comment:
Artifacts are only returned for the string types, which are being set in `ConvertValueToNomsValue`. The idea behind the artifact was that, in order to determine validity for some types, they must be processed in some way. In the case of the string types when it comes to a `UintType`, the string must be parsed into an `uint64` before its precision can be evaluated. The artifacts are just so that we're not double processing on `ConvertValueToNomsValue`. On a call to `IsValid`, we only want to check validity, so when possible I leave the conversion code in `ConvertValueToNomsValue` if I can check for validity without doing any extra processing.